frontend/src/components/charts/DurationSummaryChart.tsx

```diff
@@ -0,0 +1,128 @@
+import { useTheme } from "@mui/material/styles/";
+import ReactEcharts from "echarts-for-react";
+
+import { Difficulty } from "~shared/types/base/index";
+
+type DurationSummaryProps = {
+  durationSummary:
+    | {
+        difficulty: Difficulty;
+        timetaken: number;
+        date: Date;
+      }[]
+    | undefined;
+};
+
+export const DurationSummaryChart = ({
+  durationSummary,
+}: DurationSummaryProps) => {
+  const theme = useTheme();
+
+  const processDates = (
+    durationSummary:
+      | {
+          difficulty: Difficulty;
+          timetaken: number;
+          date: Date;
+        }[]
+      | undefined,
+  ) => {
+    const xAxis = new Set<string>();
+    if (!durationSummary) {
+      return;
+    } else {
+      // Sort the durationSummary first
+      durationSummary.sort((a, b) => a.date.getTime() - b.date.getTime());
+      // Then process it
+      durationSummary.map((duration) => {
+        const today = new Date();
+        const oneMonthAgo = new Date(
+          new Date().setMonth(new Date().getMonth() - 1),
+        );
+        const attemptDate = duration.date;
+        if (attemptDate >= oneMonthAgo && attemptDate <= today) {
+          const date =
+            attemptDate.getDate() +
+            "/" +
+            (attemptDate.getMonth() + 1) +
+            "/" +
+            attemptDate.getFullYear();
```

Comment:
Can use `date-fns` to format instead. And also `attemptDate` isn't deserialised yet.